### DISCUSSION TITLE
Server to Server calls

### DISCUSSION THREAD

**Discussion by timheerwagen** [trpc/trpc#3607] [Q&A]
I would like to establish a server to server communication. 

On server A I can create the endpoints on a server call with

`export const trpcCall = storeRouter.createCaller({});`

However, this does not work on server B, from which I also want to make a server-side call to server A.
From the client, on server B, I use the following configuration, which works great:

```
export const trpc = createTRPCNext<StoreRouter>({
  config() {
    return {
     ...
      links: [
        httpBatchLink({
          url: AServerUrl,
        }),
      ],
    ...
  },
});
```

However, I don't see any way to execute server calls with on another url.

Like for example:

```
export const trpcCall = storeRouter.createCaller({
 config() {
    return {
     ...
      left: [
        httpBatchLink({
          url: AServerUrl,
        }),
      ],
    ...
  },
});
```

**Comment by KATT** [comment]
Seems like you're describing a classic circular dependency? Maybe you need a server C where you gather the things that are used by A and B

**Reply by timheerwagen** [reply]
Even if that were the case, I wouldn't know how to fetch to another URL with tRPC using createCall as with createTRPCNext where you can set your URL at httpBatchLink.

**Reply by KATT** [reply]
Then I don't understand the question, is it about how to communicate between A & B in general? You need to use a trpc client

**Reply by timheerwagen** [reply]
First of all, thank you very much for the help so far. I am sorry if my question was not understandable.

I have a server A with Next.JS, there data is fetched using getServerSideProps and trpc.createCaller({}), this works great. Also on this server endpoints are defined, which server B should access.

Server B also has a page that should access the endpoints of server A using getServerSideProps. There, trpc.createCaller({}) is now probably the wrong choice, since this directly executes the function and does not query the endpoint of server A.

I read about createProxySSGHelpers, however I only need the data in the getServerSideProps or getStaticProps function. Later not in the client.

**Reply by KATT** [reply]
Alright, in server B you need to use a client to call server A - https://trpc.io/docs/vanilla.

You have to go over the network to call stuff on another machine - `.createCaller()` doesn't go over the network
